src/core/state/popupInstances/index.ts

```diff
@@ -153,7 +157,44 @@ export const usePopupInstanceStore = createRainbowStore<PopupInstanceStore>(
       set({ swapTokenToSell: token });
     }),
     setupPort: popupInstanceHandlerFactory(() => {
-      chrome.runtime.connect({ name: 'popup' });
+      const createPort = () => {
+        const port = chrome.runtime.connect({ name: 'popup' });
+
+        // Handle port disconnection (including BFCache scenarios)
+        port.onDisconnect.addListener(() => {
+          if (chrome.runtime.lastError) {
+            const errorMessage =
+              chrome.runtime.lastError.message || 'Unknown error';
+            console.warn('Popup port disconnected:', errorMessage);
+
+            // Check if this is the BFCache error message using utility
+            if (isBFCacheError(errorMessage)) {
+              console.log(
+                'Port disconnected due to BFCache, will reconnect on pageshow',
+              );
+              return;
+            }
+          }
+
+          // For other disconnection reasons, attempt to reconnect after a short delay
+          setTimeout(createPort, 100);
+        });
+
+        return port;
+      };
+
+      // Create initial port
+      const port = createPort();
+
+      // Handle BFCache restoration using utility
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const cleanup = onBFCacheRestore(() => {
+        // Page was restored from BFCache, recreate the port
+        createPort();
+      });
```

Comment:
Are Popup entrypoints impacted by BFCache? I was under the assumption that it would only be Inpage and Content scripts having to handle the new lifecycle